### PR TITLE
RP-1 Contract Satellites

### PR DESCRIPTION
**Pull request by ryanc55**
Remove Nav/Com/Weather payloads from SM tanks.   Introduce custom RP-1 payload satellites for NavSatPayload and ComSatPayload.

Supported by https://github.com/KSP-RO/ROLibrary/pull/20

### CODE REVIEW

File: GameData/RP-1/Parts/ContractSat/CommunicationSat.cfg [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+{
+    %TechRequired = unlockParts
+    %cost = 6
+    %entryCost = 1
+    %RP0conf = True
+    @description ^=:$: <b><color=green>From RP-1 mod</color></b>
+
+}
+
+@PART[RP-1-CommunicationSat]:FOR[xxxRP0]
```
Comment (NathanKell) [inline GameData/RP-1/Parts/ContractSat/CommunicationSat.cfg:188]: xxxRP0 is the tree pass. I wouldn't use it for adding tanks.

Comment (NathanKell) [reply GameData/RP-1/Parts/ContractSat/CommunicationSat.cfg:188]: (I don't actually get why this isn't part of the base PART tbh)

File: Source/UpgradeScripts/v2_10_1RemovePayload.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        public override Version EarliestCompatibleVersion { get => new Version(2, 0, 0); }
+        protected static Version _targetVersion = new Version(2, 10, 1);
+        public override Version TargetVersion => _targetVersion;
+
+        private TestResult TestRecurse(ConfigNode node)
+        {
+            ConfigNode.Value v = null;
+            if (node.name == "RESOURCE" && (v = node.values.values.Find(s => s.name == "name")) != null)
+            {
+                if (v.value == "ComSatPayload" || v.value == "NavSatPayload")
```
Comment (NathanKell) [inline Source/UpgradeScripts/v2_10_1RemovePayload.cs:22]: Weathersat payload is removed above but there's no check for it here?

File: Source/UpgradeScripts/v2_10_1RemovePayload.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                {
+                    return TestResult.Upgradeable;
+                }
+            }
+
+            foreach (ConfigNode n in node.nodes)
+            {
+                var res = TestRecurse(n);
```
Comment (NathanKell) [inline Source/UpgradeScripts/v2_10_1RemovePayload.cs:30]: This won't actually find MFT blocks, because their node name is TANK not RESOURCE, no?

Comment (NathanKell) [reply Source/UpgradeScripts/v2_10_1RemovePayload.cs:30]: (and if that's not your intention here, what _are_ you trying to find by recursing down every node in the PART node?)

File: Source/UpgradeScripts/v2_10_1RemovePayload.cs (hunk 2 above)
Comment (ryanc55) [reply Source/UpgradeScripts/v2_10_1RemovePayload.cs:22]: Thanks, I'll add that check.

**Comment by NathanKell** [comment]
Closed?

**Comment by ryanc55** [comment]
Yeah, I messed it up.  I committed a merge commit from master to my fork and made the diffs very messy.  I'll open a new one tonight with the above changes and the new models.

**Pull request closed without merging**